policyengine_api/services/ai_analysis_service.py

```diff
@@ -0,0 +1,92 @@
+import anthropic
+import os
+import json
+import time
+from typing import Generator
+from policyengine_api.data import local_database
+
+
+class AIAnalysisService:
+    """
+    Base class for various AI analysis-based services,
+    including SimulationAnalysisService, that connects with the analysis
+    local database table
+    """
+
+    def get_existing_analysis(
+        self, prompt: str
+    ) -> Generator[str, None, None] | None:
```

Comment:
and `from typing import Optional`
```suggestion
    ) -> Optional[Generator[str, None, None]]:
```